Update support team process (#311)
2i2c-org/team-compass: projects/managed-hubs/support.md

Support team issue
```

## Responsibilities and expectations

These are some major responsibilities and goals of the support team:

- Be the main point of contact for all support tickets regardless of the community
- Communicate with the person opening a support ticket, to keep them in the loop
- Create backlog issues as-needed
- Route support issues to the engineering team in order to make the changes necessary to resolve the issue.
- During events, be extra attentive to support channels in case a change is needed.

:::{note}
We should try to respond to all support-related communications **within one working day**.
Even if the support request is not resolved, others appreciate when we keep them in-the-loop about what we're doing.
:::

## Tracking support items

### FreshDesk requests

For bugs, hub changes, improvements, and general questions, we use the [e-mail address `support@2i2c.org`](mailto:support@2i2c.org).
This is connected with [the 2i2c FreshDesk account](https://2i2c.freshdesk.com/), and is a central place for all support-related requests.

Many requests may be immediately resolved via a quick action or response to the community representative.

> However, if a request requires consultation with the team, open a backlog issue.

Author: ```suggestion
However, if a request requires consultation with the team, first open a thread in the {guilabel}`#support-freshdesk` Slack channel, and if necessary open a backlog issue or a .
```

(support:issues)=
### Support issues

If resolving a support issue requiries ongoing or concerted owrk, open a backlog issue tagged with the {guilabel}`support` label.

Support backlog items should:

- Be placed in our [Sprint Board](coordination:sprint-board).
- Include a reference to any FreshDesk tickets if they exist.
- Be prioritized over other backlog items for that sprint.

## Communication channels

We use two Slack channels to discuss support-related questions on the 2i2c hub:

- The [#support-freshdesk](https://2i2c.slack.com/archives/C028WU9PFBN) is for discussing support tickets on Freshdesk.
  Open a Slack thread for each support ticket, to have conversation around it amongst the team.
  This channel is private because there’s an assumption of privacy when we discuss community-specific support matters.
- The [#hub-support](https://2i2c.slack.com/archives/C01DB2JRP8W) channel is for discussing more general support-related issues relating to operating the 2i2c Hubs, and not not related to a specific FreshDesk ticket.
  This channel is public, so can be useful if we wish to invite conversation from non 2i2c team members.

## Lifecycle of a support request

> - When you receive a ticket, communicate that you have received it and note the next steps you’ll follow to resolve it (even if this just means investigating because you're not sure what is wrong).

Author: ```suggestion
- We should try to respond to all support-related communications within one working day.
- When you receive a ticket, communicate that you have received it and note the next steps you’ll follow to resolve it (even if this just means investigating because you're not sure what is wrong).
```

> - If needed, open a [support issue and add it to the backlog](support:issues).

Reviewer: There's one thing that I've been asking myself. Should we link the people opening support tickets to the relevant GitHub issues?

Author: What do you think? I can see an argument both ways. If we gave a link to the issue then:

- PRO: there is more visibility and maybe more efficient communication since they could participate directly in the issue where work is being done.
- CON: we've now got two communication channels with the person for this incident - one in FreshDesk and one in GitHub. Moreover we're not guaranteed that their communication in GitHub will actually be helpful...

Reviewer 2: NO... if people are interested enough they will find the issue and jump into it, but our canonical tunnel should be Freshdesk!

Support team issue
```

## Responsibilities and expectations

These are some major responsibilities and goals of the support team:

- Be the main point of contact for all support tickets regardless of the community
- Communicate with the person opening a support ticket, to keep them in the loop
- Create backlog issues as-needed
- Route support issues to the engineering team in order to make the changes necessary to resolve the issue.
- During events, be extra attentive to support channels in case a change is needed.

:::{note}
We should try to respond to all support-related communications **within one working day**.
Even if the support request is not resolved, others appreciate when we keep them in-the-loop about what we're doing.
:::

## Tracking support items

### FreshDesk requests

For bugs, hub changes, improvements, and general questions, we use the [e-mail address `support@2i2c.org`](mailto:support@2i2c.org).
This is connected with [the 2i2c FreshDesk account](https://2i2c.freshdesk.com/), and is a central place for all support-related requests.

Many requests may be immediately resolved via a quick action or response to the community representative.

> However, if a request requires consultation with the team, first open a thread in the {guilabel}`#support-freshdesk` Slack channel, and if necessary open a backlog issue or a .

Reviewer 2: Incomplete sentence?

> However, if a request requires consultation with the team, first open a thread in the {guilabel}`#support-freshdesk` Slack channel, and if necessary open a backlog issue or a .

Author: ```suggestion
If a request requires consultation with the team, then:

- Open a Slack thread in the {guilabel}`#support-freshdesk` channel to discuss with others. If we need to track a work item to resolve the ticket over time, then:
- Open a GitHub issue with the {guilabel}`support` label, and add it to our Sprint backlog.
```

We use two Slack channels to discuss support-related questions on the 2i2c hub:

- The [#support-freshdesk](https://2i2c.slack.com/archives/C028WU9PFBN) is for discussing support tickets on Freshdesk.
  Open a Slack thread for each support ticket, to have conversation around it amongst the team.
  This channel is private because there’s an assumption of privacy when we discuss community-specific support matters.
- The [#hub-support](https://2i2c.slack.com/archives/C01DB2JRP8W) channel is for discussing more general support-related issues relating to operating the 2i2c Hubs, and not not related to a specific FreshDesk ticket.
  This channel is public, so can be useful if we wish to invite conversation from non 2i2c team members.

## Lifecycle of a support request

Below is a summary of the major steps that make up the support process, from beginning to end of a ticket.

:::{note}
We should try to respond to all support-related communications within one working day.
:::

- When you receive a ticket, communicate that you have received it and note the next steps you’ll follow to resolve it (even if this just means investigating because you're not sure what is wrong).

> - If needed, open a [support issue and add it to the backlog](support:issues).

Author: ```suggestion
- If needed, open a [support issue and add it to the backlog](support:issues).
  Note that we should still use Freshdesk for support communication on this request, the GitHub issue is just for 2i2c team members to coordinate.
```

- Identify an engineering team member that can pick up this work.
- Give the user updates every day or two as we work to resolve the issue.
- When it is resolved, tell the user what you’ve done to resolve the issue.
- Close any FreshDesk tickets and support issues that are related to this request.

## How to prioritize support requests

> - Ask whether an issue is "critical" (AKA does the hub not work at all) and prioritize these if so

Reviewer 2: You can't ask if the issue is critical, IMHO.
People will tend to say, yes, it is critical... we should gather as much info as we can about the issue and decide if it is actually critical or not.

Author: Good point - I think here I actually meant more like "Ask yourself" rather than "Ask the other person" - will clarify this.

> - Ask whether an issue is "critical" (AKA does the hub not work at all) and prioritize these if so

Author: ```suggestion
- Ask questions that will help us understand whether an issue is "critical". For example, "Does this effect many users on the hub, or just a subset?", "Does this problem always happen, or intermittently?", "Is the hub accessible at all or is it totally down?".
```
